lib/utils.js

```diff
@@ -538,6 +538,23 @@ exports.populate = function populate(path, select, model, match, options, subPop
     throw new TypeError('utils.populate: invalid path. Expected string. Got typeof `' + typeof path + '`');
   }
 
+  if (Array.isArray(subPopulate)) {
+    let ret = [];
+    subPopulate.forEach(function(obj) {
+      if (/[\s]/.test(obj.path)) {
+        let copy = exports.object.shallowCopy(obj);
+        let paths = copy.path.split(' ');
+        paths.forEach(function(p) {
+          copy.path = p;
```

Comment:
Do you need to clone this?

Reply:
because the select, populate, and match options are all going to be the same for each one, I can just duplicate the obj once ( line 545 ) and reuse it to call populate on the same object  with a new path each time.